### PR TITLE
Support ALTER MODIFY SETTING for Memory tables

### LINKED ISSUE #61922: Support ALTER MODIFY SETTING for Memory tables
**Use case**

```
milovidov-desktop :) ALTER TABLE numbers_capped MODIFY SETTING max_rows_to_keep = 100_000_000

ALTER TABLE numbers_capped
    (MODIFY SETTING max_rows_to_keep = 100000000)

Query id: 2a59cea3-d061-4343-b25e-6e51b106156e


Elapsed: 0.012 sec. 

Received exception from server (version 24.3.1):
Code: 36. DB::Exception: Received from localhost:9000. DB::Exception: Cannot alter settings, because table engine doesn't support settings changes. (BAD_ARGUMENTS)
```

### PR DESCRIPTION
**Pull request by zhongyuankai**
close #61922
### Changelog category (leave one):
- New Feature

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Modifying memory table settings through `ALTER MODIFY SETTING` is now supported.
```
ALTER TABLE memory MODIFY SETTING min_rows_to_keep = 100, max_rows_to_keep = 1000;
```

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

---
### Modify your CI run:
**NOTE:** If your merge the PR with modified CI you **MUST KNOW** what you are doing
**NOTE:** Checked options will be applied if set before CI RunConfig/PrepareRunConfig step

#### Include tests (required builds will be added automatically):
- [ ]  Fast test
- [ ]  Integration Tests
- [ ]  Stateless tests
- [ ]  Stateful tests
- [ ]  Unit tests
- [ ]  Performance tests
- [ ]  All with ASAN
- [ ]  All with TSAN
- [ ]  All with Analyzer
- [ ]  Add your option here

#### Exclude tests:
- [ ]  Fast test
- [ ]  Integration Tests
- [ ]  Stateless tests
- [ ]  Stateful tests
- [ ]  Performance tests
- [ ]  All with ASAN
- [ ]  All with TSAN
- [ ]  All with MSAN
- [ ]  All with UBSAN
- [ ]  All with Coverage
- [ ]  All with Aarch64
- [ ]  Add your option here

#### Extra options:
- [ ]  do not test (only style check)
- [ ]  disable merge-commit (no merge from master before tests)
- [ ]  disable CI cache (job reuse)

#### Only specified batches in multi-batch jobs:
- [ ]  1
- [ ]  2
- [ ]  3
- [ ]  4

### CODE REVIEW

**Comment by robot-clickhouse-ci-2** [comment]
*This is an automated comment for commit f1ae99b1136218a1ff8cd281293868b7fdabf9fe with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/62039/f1ae99b1136218a1ff8cd281293868b7fdabf9fe/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>PR Check</td><td>There's no description for the check yet, please add it to tests/ci/ci_config.py:CHECK_DESCRIPTIONS</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

**Review by yariks5s — COMMENTED** [review]
Interesting to add some cases in tests:
- when initially we have `max_bytes_to_keep`/`max_rows_to_keep` = x and we set `max_bytes_to_keep`/`max_rows_to_keep` = y, y < x. Good to check if everything is working right and we have a smaller circular buffer.
- when initially we have a memory table with no settings and we apply these settings to this table.
- when initially we don't have `min_bytes_to_keep`/`min_rows_to_keep` settings for the table and add them with `ALTER`. Also good to check if min settings are changed correctly
- when we try to set incorrect values for those settings

**Comment by zhongyuankai** [comment]
Related tests have been added.

File: tests/queries/0_stateless/03032_storage_memory_modify_settings.sql [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+INSERT INTO memory SELECT * FROM numbers(0, 100);
+INSERT INTO memory SELECT * FROM numbers(100, 1000);
+SELECT total_bytes FROM system.tables WHERE name = 'memory' and database = currentDatabase();
+
+ALTER TABLE memory MODIFY SETTING min_rows_to_keep = 100, max_rows_to_keep = 1000;
+SELECT total_bytes FROM system.tables WHERE name = 'memory' and database = currentDatabase();
+
+INSERT INTO memory SELECT * FROM numbers(1000, 500);
+SELECT total_bytes FROM system.tables WHERE name = 'memory' and database = currentDatabase();
+
```
[the remark this answers is not in the document]
Comment (yariks5s) [reply tests/queries/0_stateless/03032_storage_memory_modify_settings.sql]: FYI fixed this problem but still needs clarification

**Review by yariks5s — APPROVED** [review]

**Pull request merged**